### PR TITLE
Vercel setup

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Vercel deploy

Screen Shot 2022-11-01 at 11 02 51

- Added a test env job that should be removed before merging
- Added a `Vercel dev` job that will trigger when there's a commit to `develop` branch
It builds and publishes to Vercel `cowswap-dev` project "production" env: https://cowswap-dev-cowswap.vercel.app/

- Added a `Vercel pre-prod` job that will trigger when there's a commit to `develop` branch
It builds and publishes in parallel to both `cowswap-barn` and `cowswap-staging` projects "production" envs: https://cowswap-staging-cowswap.vercel.app/ and https://cowswap-barn-cowswap.vercel.app/
**Note**: There are no deployments for the above as the workflow hasn't triggered yet :)

**Note**: Prod not setup yet as it'll be triggered manually. Will be another PR

# Testing

1. Check the build steps
2. There is a special job ONLY for testing in this PR (`Test deploy...`)
3. It builds and publishes to Vercel `cowswap-dev` project "production" env: https://cowswap-dev-cowswap.vercel.app/

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
 
       - name: Set output of cache
         id: yarn-cache
-        run: echo "::set-output name=dir::$(yarn cache dir)"
+        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
```
Comment (anxolin) [inline .github/workflows/ci.yml:39]: it was not working correctly before? This fixes some issue? Was this tested?

**Review by anxolin — APPROVED** [review]
Looks great! good job

Question, after the rename, shouldn't it be the name https://swap-dev-cow.vercel.app/#/swap?chain=mainnet

File: .github/workflows/ci.yml (hunk 1 above)
Comment (alfetopito) [reply .github/workflows/ci.yml:39]: > it was not working correctly before? 

Yes it was

> This fixes some issue? 

That syntax is deprecated
Screen Shot 2022-11-02 at 13 55 31

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/


> Was this tested?

Yep, I use this elsewhere and nothing changed in the actual behaviour

**Comment by alfetopito** [comment]
> Looks great! good job
> 
> Question, after the rename, shouldn't it be the name https://swap-dev-cow.vercel.app/#/swap?chain=mainnet

I think it'll update on new deployments only.

For example, on PRs open before the change and that had builds after the change you can see multiple entries in the Vercel comment:

Screen Shot 2022-11-02 at 13 57 09

**Pull request merged**